src/application/rounder.py

```diff
@@ -0,0 +1,139 @@
+
+        7. Is fallback for sigfigs given (not -1) (see config)?
+           Round value according to number of sigfigs.
+
+        8. Is fallback for decimal places given (not -1) (see config)?
+           Round value according to number of decimal places.
+
+        TODO: Warning message if user specifies exact value and sigfigs etc.
+        """
+        cls._round_result(result, config)
+
+        short = result.get_short_result()
+        if short:
+            cls._round_result(short, config)
+
+    @classmethod
+    # pylint: disable-next=too-many-branches
+    def _round_result(cls, result: Result, config: RoundingConfig) -> None:
+        """See the docstring of the public `round_result` for details."""
+
+        value = result.value
+        uncertainties = result.uncertainties
+
+        # Rounding hierarchy 1:
+        if value.is_exact():
+            cls._uncertainties_set_min_exponents(uncertainties, value.get_min_exponent())
+
+        # Rounding hierarchy 2:
+        elif result.sigfigs is not None:
+            value.set_sigfigs(result.sigfigs)
+            cls._uncertainties_set_min_exponents(uncertainties, value.get_min_exponent())
+
+        # Rounding hierarchy 3:
+        elif result.decimal_places is not None:
+            value.set_min_exponent(-result.decimal_places)
+            cls._uncertainties_set_min_exponents(uncertainties, value.get_min_exponent())
+
+        # Rounding hierarchy 4:
+        elif config.sigfigs > -1:
+            value.set_sigfigs(config.sigfigs)
+            cls._uncertainties_set_min_exponents(uncertainties, value.get_min_exponent())
+
+        # Rounding hierarchy 5:
+        elif config.decimal_places > -1:
+            min_exponent = -config.decimal_places
+            value.set_min_exponent(min_exponent)
+            cls._uncertainties_set_min_exponents(uncertainties, min_exponent)
+
+        # Rounding hierarchy 6:
+        elif len(uncertainties) > 0:
+            for u in uncertainties:
+                if u.uncertainty.is_exact():
+                    continue
+
+                normalized_value = abs(u.uncertainty.get()) * 10 ** (
+                    -Helpers.get_exponent(u.uncertainty.get())
+                )
+
+                if round(normalized_value, 1) >= 3.0:
+                    u.uncertainty.set_sigfigs(1)
+                else:
+                    u.uncertainty.set_sigfigs(2)
+
+            min_exponent = min(u.uncertainty.get_min_exponent() for u in uncertainties)
+            value.set_min_exponent(min_exponent)
+
+        # Rounding hierarchy 7:
+        elif config.sigfigs_fallback > -1:
+            value.set_sigfigs(config.sigfigs_fallback)
+
+        # Rounding hierarchy 8:
+        elif config.decimal_places_fallback > -1:
+            min_exponent = -config.decimal_places_fallback
+            value.set_min_exponent(min_exponent)
+
+        else:
+            raise RuntimeError  # TODO: add message (!)
```

Comment:
@paul019 Please add a message here for the `RuntimeError`. In which cases could this even happen?